hyperspy/drawing/markers.py

```diff
@@ -0,0 +1,816 @@
+        self.transform = transform
+
+        # Events
+        self.events = Events()
+        self.events.closed = Event(
+            """
+            Event triggered when a marker is closed.
+
+            Arguments
+            ---------
+            marker : Marker
+                The marker that was closed.
+            """,
+            arguments=["obj"],
+        )
+        self._closing = False
+
+    @property
+    def _is_iterating(self):
+        if self._plot_on_signal:
+            return np.any([is_iterating(value) for key, value in self.kwargs.items()])
+        else:  # currently iterating navigation markers arenot supported
+            return False
+
+    def _get_transform(self, attr="_transform"):
+        if self.ax is not None:  # return the transform
+            transforms = {
+                "data": self.ax.transData,
+                "axes": self.ax.transAxes,
+                "display": IdentityTransform(),
+                "yaxis": self.ax.get_yaxis_transform(),
+                "xaxis": self.ax.get_xaxis_transform(),
+                "relative": self.ax.transData,
+            }
+            return transforms[getattr(self, attr)]
+        else:  # return the string value
+            return getattr(self, attr)
+
+    def _set_transform(self, value, attr="_transform"):
+        arg_list = ["data", "axes", "xaxis", "yaxis", "display", "relative"]
+        if value not in arg_list:
+            str_ = ", ".join([f"`{v}`" for v in arg_list])
+            raise ValueError(
+                f"The transform must be one of {str_}."
+            )
+        setattr(self, attr, value)
+        if self.collection is not None and self.ax is not None:
+            getattr(self.collection, f"set{attr}")(getattr(self, attr[1:]))
+            # Update plot
+            self.update()
+
+    @property
+    def offset_transform(self):
+        return self._get_transform(attr="_offset_transform")
+
+    @offset_transform.setter
+    def offset_transform(self, value):
+        self._set_transform(value, attr="_offset_transform")
+
+    @property
+    def transform(self):
+        return self._get_transform(attr="_transform")
+
+    @transform.setter
+    def transform(self, value):
+        self._set_transform(value, attr="_transform")
+
+    def __len__(self):
+        """
+        Return the number of markers in the collection at the current
+        navigation coordinate.
+        """
+        if self._is_iterating and self.axes_manager is None:
+            # with variable length markers, the axes_manager is needed to
+            # know the navigation coordinates of the signal
+            raise RuntimeError(
+                "Variable length markers must have been plotted to provide "
+                "the numbers of markers at the current navigation coordinates."
+                )
+
+        return list(self.get_data_position().values())[0].shape[0]
+
+    def delete_index(self, keys, index):
```

Comment:
The `delete_index` and `append_kwargs` should have symmetrical name? Maybe `remove_item` and `add_item`? 

How does it work with variable length markers? It seems that it removes for all navigation position, but I would imagine that it would be useful to deleted an item from a specific navigation position?

Reply:
Name change is a good idea, I like remove_item and add_item! 

Deleting an item from a specific navigation position is a good idea but the `Markers` class already doesn't have a great way to visualize the underlying kwargs when the kwargs are ragged.  I kind of feel like while there are cases were I might want to say exclude some peak that is found, it would be better to handle that in the signal (like in pyxem) where we have operations to exclude peaks with some magnititude.